### PR TITLE
[Transforms] SVTR transforms

### PR DESCRIPTION
**Pull request by gaotongxiao**

### CODE REVIEW

File: mmocr/datasets/transforms/textrecog_transforms.py [hunk 1]
```diff
@@ -251,3 +252,447 @@ def __repr__(self) -> str:
...
+
+    def __repr__(self) -> str:
+        repr_str = self.__class__.__name__
+        repr_str += f'(min_pixels = {self.min_pixels}, '
+        repr_str += f'max_pixels = {self.max_pixels})'
+        return repr_str
+
+
+@TRANSFORMS.register_module()
+class ImageContentJitter(BaseTransform):
```
Comment (Harold-lkk) [inline mmocr/datasets/transforms/textrecog_transforms.py:627]: Maybe rename it to ImageRightBottomShift

File: mmocr/datasets/transforms/textrecog_transforms.py [hunk 2]
```diff
@@ -251,3 +252,447 @@ def __repr__(self) -> str:
...
+            results (dict): Result dict from loading pipeline.
+
+        Returns:
+            dict: Jittered results.
+        """
+        h, w = results['img'].shape[:2]
+        img = results['img'].copy()
+        if h > 10 and w > 10:
+            thres = min(h, w)
+            jitter_range = int(random.random() * thres * 0.01)
```
Comment (Harold-lkk) [inline mmocr/datasets/transforms/textrecog_transforms.py:655]: can add a parameter for 0.01

File: mmocr/datasets/transforms/textrecog_transforms.py [hunk 3]
```diff
@@ -251,3 +252,447 @@ def __repr__(self) -> str:
...
+                                                     src[nyi1, nxi], src[nyi1,
+                                                                         nxi1])
+
+        dst = np.clip(dst, 0, 255)
+        dst = np.array(dst, dtype=np.uint8)
+
+        return dst
+
+    @staticmethod
+    def _bilinear_interp(x, y, v11, v12, v21, v22):
```
Comment (Harold-lkk) [inline mmocr/datasets/transforms/textrecog_transforms.py:555]: docstring

Comment (Harold-lkk) [inline mmocr/datasets/transforms/textrecog_transforms.py:555]: Maybe add TODO to split it into utils

File: mmocr/datasets/transforms/textrecog_transforms.py [hunk 4]
```diff
@@ -251,3 +252,447 @@ def __repr__(self) -> str:
...
+                cut * cut_idx + np.random.randint(thresh) - half_thresh,
+                img_h + np.random.randint(thresh) - half_thresh
+            ])
+
+        dst = self.warp_mls(img, src_pts, dst_pts, img_w, img_h)
+
+        return dst
+
+    def tia_stretch(self, img: np.ndarray, segment: int = 4) -> np.ndarray:
+        """Image stretching."""
```
Comment (Harold-lkk) [inline mmocr/datasets/transforms/textrecog_transforms.py:348]: can add TODO to add  parameter docs

File: mmocr/datasets/transforms/textrecog_transforms.py [hunk 5]
```diff
@@ -251,3 +252,447 @@ def __repr__(self) -> str:
...
+        """Call function to pad images.
+
+        Args:
+            results (dict): Result dict from loading pipeline.
+
+        Returns:
+            dict: Updated result dict.
+        """
+        h, w = results['img'].shape[:2]
+        if h >= 20 and w >= 20:
```
[the remark this answers is not in the document]
Comment (gaotongxiao) [reply mmocr/datasets/transforms/textrecog_transforms.py:287]: These are the minimum h and w requirements

File: docs/en/api/transforms.rst [hunk 6]
```diff
@@ -50,6 +49,10 @@ TextRecog Transforms
    :nosignatures:
    :template: classtemplate.rst
 
+   TextRecogGeneralAug
+   TextRecogRandomCrop
```
Comment (Harold-lkk) [inline docs/en/api/transforms.rst:53]: CropHeight

**Review by Harold-lkk — APPROVED** [review]

**Pull request merged**